playground/src/Editor/settings.ts

```diff
@@ -22,29 +23,33 @@ export function stringify(settings: Settings): string {
 /**
  * Persist the configuration to a URL.
  */
-export async function persist(settingsSource: string, pythonSource: string) {
-  const hash = lzstring.compressToEncodedURIComponent(
-    settingsSource.replaceAll("$$$", "$$$$$$") + "$$$" + pythonSource,
-  );
-  await navigator.clipboard.writeText(
-    window.location.href.split("#")[0] + "#" + hash,
-  );
+export async function persist(
+  settingsSource: string,
+  pythonSource: string,
+): Promise<void> {
+  const id = uuidv4();
```

Comment:
Perhaps this should instead be a truncated hex string of the hash, so that you get a stable URL for a given snippet?

Reply:
What about hash collisions? 

I guess it depends on the desired UX:

* Should users be able to keep making changes to the linked version even after sharing (and support different versions)
* or do we want unique URLs for each possible playgorund.

Reply:
I'd expect the second one for now.

Reply:
I worry about truncated hash collisions though?

Reply:
Yeah I think the second one is the correct model. It would perhaps be nice if hitting `Share` multiple times on the same snippet gave you the same URL though, that's what I meant by "Stable URL" (same content leads to same URL).

Reply:
What I like about [astexplorer.net ](https://astexplorer.net/) is that the workflow is explicit about when it creates a new gist vs updating the existing one.

The [Rust playground](https://play.rust-lang.org/) creates a new gist whenever you hit save. (Gists everywhere)